common/beerocks/bcl/include/bcl/network/socket.h

```diff
@@ -141,4 +145,35 @@ class SocketSelect {
     std::string m_error;
 };
 
+class SocketPoll {
```

Comment:
Originally select was chosen as it's somewhat more cross platform (from the days beerocks ran on Windows).
Since we now support Linux based platforms only, I think we should replace select with epoll.